File: exercise1/core/src/com/hermannm/helicopter/Helicopter.java
package com.hermannm.helicopter;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.hermannm.helicopter.states.GameStateManager;
import com.hermannm.helicopter.states.MenuState;

public class Helicopter extends ApplicationAdapter {
	public static final int WIDTH = 800;
	public static final int HEIGHT = 480;
	public static final String TITLE = "Helicopter";
	private GameStateManager stateManager;
	private SpriteBatch batch;
	Texture img;
	
	@Override
	public void create () {
		batch = new SpriteBatch();
		stateManager = new GameStateManager();
		Gdx.gl.glClearColor(1, 0, 0, 1);
		stateManager.push(new MenuState(stateManager));
	}

	@Override
	public void render () {
		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
		stateManager.update(Gdx.graphics.getDeltaTime());
		stateManager.render(batch);
	}
	
	@Override
	public void dispose () {
		batch.dispose();
		img.dispose();
	}
}

File: exercise1/core/src/com/hermannm/helicopter/Helicopter.kt
package com.hermannm.helicopter

import com.badlogic.gdx.ApplicationAdapter
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.hermannm.helicopter.states.GameStateManager
import com.hermannm.helicopter.states.MenuState

class Helicopter : ApplicationAdapter() {
    companion object {
        const val WIDTH = 800F
        const val HEIGHT = 480F
        const val TITLE = "Helicopter"
    }
    private var batch: SpriteBatch? = null
    private var stateManager: GameStateManager? = null
    override fun create() {
        batch = SpriteBatch()
        stateManager = GameStateManager()
        Gdx.gl.glClearColor(1F, 0F, 0F, 1F)
        stateManager?.push(MenuState(stateManager!!))
    }
    override fun render() {
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)
        stateManager?.update(Gdx.graphics.getDeltaTime())
        stateManager?.render(batch!!)
    }

    override fun dispose() {
        batch?.dispose()
    }
}